native/src/routes/jpal-tracking/components/JpalTracking.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import React, { useEffect, useState } from 'react'
+
+import Caption from '../../../modules/common/components/Caption'
+import type { ThemeType } from 'build-configs/ThemeType'
+import { Switch, Text, TextInput, View } from 'react-native'
+import type { TFunction } from 'react-i18next'
+import type { StateType } from '../../../modules/app/StateType'
+import AppSettings, { defaultSettings } from '../../../modules/settings/AppSettings'
+import type { NavigationPropType, RoutePropType } from '../../../modules/app/constants/NavigationTypes'
+import type { JpalTrackingRouteType } from 'api-client'
+import styled from 'styled-components/native'
+import LayoutContainer from '../../../modules/layout/containers/LayoutContainer'
+
+const ThemedText = styled.Text`
+  display: flex;
+  text-align: left;
+  color: ${props => props.theme.colors.textColor};
+  font-family: ${props => props.theme.fonts.decorativeFontRegular};
+  padding: 10px 0;
+`
+
+const DescriptionContainer = styled.View`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 15px 0 5px;
+`
+
+const Input = styled(TextInput)`
+  padding: 15px;
+  border-width: 1px;
+  border-color: ${props => props.theme.colors.themeColor};
+  text-align-vertical: top;
+  height: 50px;
+`
+
+export type PropsType = {|
+  theme: ThemeType,
+  t: TFunction,
+  route: RoutePropType<JpalTrackingRouteType>,
+  navigation: NavigationPropType<JpalTrackingRouteType>
+|}
+
+const JpalTracking = (props: PropsType, state: StateType) => {
+  const [appSettings] = useState(new AppSettings())
+  const [settings, setSettings] = useState(defaultSettings)
+  const [settingsLoaded, setSettingsLoaded] = useState(false)
+  const [settingsUpdated, setSettingsUpdated] = useState(false)
+  const [displayedTrackingCode, setDisplayedTrackingCode] = useState('')
+  const routeTrackingCode = props.route.params.trackingCode
+
+  useEffect(() => {
+    if (routeTrackingCode) {
+      appSettings.setJpalTrackingCode(routeTrackingCode)
+    }
+  }, [appSettings, routeTrackingCode])
+
+  useEffect(() => {
+    const loadSettings = async () => {
+      try {
+        const loadedSettings = await appSettings.loadSettings()
+        setSettingsLoaded(true)
+        setSettingsUpdated(false)
+        setSettings(loadedSettings)
+        setDisplayedTrackingCode(loadedSettings.jpalTrackingCode)
+      } catch (e) {
+        console.error('Failed to load settings.')
+      }
+    }
+    loadSettings()
+  }, [appSettings, settingsUpdated])
```

Comment:
I don't think we have to reload the settings every time we change something. I think it is enough to just load the settings once initially and keep only relevant settings in a local state. Then you could just catch possible errors during setting of the tracking code or tracking enabled to the global app settings and change the local state if everything succeeds and no error is thrown. Than you can safely assume that the local state always matches the global state.

Reply:
done